test/test_transformers.py

```diff
@@ -2207,6 +2207,19 @@ def test_cudnn_attention_different_dk_dv(self, device):
 
         self.assertEqual(actual.contiguous(), math_ref.contiguous().to(dtype), atol=1e-3, rtol=1e-2)
 
+    @skipIfRocm  # No cuDNN Attention
+    @unittest.skipIf(not PLATFORM_SUPPORTS_CUDNN_ATTENTION, "cuDNN Attention is not supported on this system")
+    def test_cudnn_attention_fail_d128(self, device):
+        # Test that cuDNN attention dispatching correctly bails out on d > 128
+        b, h = 8, 2
+        s_q, s_kv = 128, 128
+        d_qk, d_v = 128, 144
+
+        q = torch.randn(b, h, s_q, d_qk, device=device, dtype=torch.bfloat16)
+        k = torch.randn(b, h, s_kv, d_qk, device=device, dtype=torch.bfloat16)
+        v = torch.randn(b, h, s_kv, d_v, device=device, dtype=torch.bfloat16)
+
```

Comment:
Can you also use the context manger to ensure that cudnn is picked here

Reply:
We actually want the opposite here, basically we're checking that the cuDNN frontend doesn't cause an exception because `can_use_cudnn_attention` correctly decides that this case is unsupported

Reply:
Yeah that makes sense! I think how you structured the test below is right, Force it only be able to run the cudnn kernel and if thats the case we except a hard runtime error in this case